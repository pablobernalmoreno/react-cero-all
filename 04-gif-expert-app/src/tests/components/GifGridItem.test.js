import React from "react";
import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";

describe("Pruebas en GifGridItem", () => {
  const title = "Titulo";
  const url = "https://";
  const wrapper = shallow(<GifGridItem title={title} url={url} />);
  test("debe mostrar el componente correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("debe contener un parrafo con el 'title'", () => {
    const p = wrapper.find("p");
    expect(p.text().trim()).toBe(title);
  });

  test("debe de tener la imagen igual 'url' y 'alt' de los props", () => {
    const img = wrapper.find("img");
    expect(img.prop("src")).toBe(url);
    expect(img.prop("alt")).toBe(title);
  });

  test("debe tener 'animate__bounceOut'", () => {
    const div = wrapper.find("div");
    expect(div.prop("className").includes("animate__bounceOut")).toBe(true);
  });
});
