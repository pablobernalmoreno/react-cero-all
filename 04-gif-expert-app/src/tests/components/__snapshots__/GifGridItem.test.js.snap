// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridItem debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__bounceOut"
>
  <img
    alt="Titulo"
    src="https://"
  />
  <p>
    Titulo
  </p>
</div>
`;
